### PR TITLE
connecting Panacea and Asclepius

### PR DESCRIPTION
**Pull request by c-brenn**
outline of how Panacea will depend on and interact with Asclepius.

### CODE REVIEW

File: panacea/test/plugs/require_asclepius_test.exs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      Panacea.Asclepius.set_readiness(false)
+
+      conn = conn |> RequireAsclepius.call([])
+
+      assert conn.status == 503
+      assert conn.halted
+    end
+  end
+
+  describe "whe asclepius is available" do
```
Comment (houli) [inline panacea/test/plugs/require_asclepius_test.exs]: "when Asclepius"

File: panacea/lib/panacea/asclepius/remote.ex [hunk 2]
```diff
@@ -0,0 +1,17 @@
+defmodule Panacea.Asclepius.Remote do
+  @asclepius_uri Keyword.get(Application.get_env(:panacea, :asclepius), :uri)
+  @ping_timeout 50
+
+  def ping do
+    asclepius_uri("/ping")
+    |> HTTPoison.get([recv_timeout: @ping_timeout])
+    |> case() do
```
Comment (houli) [inline panacea/lib/panacea/asclepius/remote.ex]: `case HTTPoison.get(asclepius_uri("/ping"), [recv_timeout: @ping_timeout]) do`
probably easier to read here

**Comment by 22a** [comment]
Brilliant work, love the /asclepius pipeline! 👹 closes #48

**Comment by c-brenn** [comment]
A bit going on here.

Adds a plug to handle requests that need Asclepius to be booted

Adds modules to abstract making requests to Asclepius -> Http and Mock
The module used is configured per environment - see `config :panacea, :asclepius_api`
This should make it very easy to test controllers that need to interact with asclepius.
A behaviour module is used to ensure Http and Mock implement the same contract

File: panacea/lib/panacea/asclepius/remote/http.ex [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  @asclepius_uri Keyword.get(Application.get_env(:panacea, :asclepius), :uri)
+  @ping_timeout 50
+  # one minute timeout for now
+  @ddi_timeout 60 * 1000
+
+  def ping do
+    asclepius_uri("/ping")
+    |> HTTPoison.get([recv_timeout: @ping_timeout])
+    |> case() do
```
Comment (houli) [inline panacea/lib/panacea/asclepius/remote/http.ex]: I think this was nicer and easier to understand with just `case HTTPoison.get(asclepius_uri("/ping"), [recv_timeout: @ping_timeout])`

**Review by houli — CHANGES_REQUESTED** [review]

File: panacea/lib/panacea/asclepius/remote/http.ex (hunk 3 above)
Comment (c-brenn) [reply panacea/lib/panacea/asclepius/remote/http.ex]: Might change it to pipe into `decode_response` instead.
Then all of the `Remote` callbacks will have the same return type -> `{:ok, response} | {:error, reason}`.
Callers can then choose to ignore or use the response/error

Comment (houli) [reply panacea/lib/panacea/asclepius/remote/http.ex]: 👵 knows best

**Review by houli — APPROVED** [review]

**Pull request merged**